### PR TITLE
Issue 52

### PR DESCRIPTION
**Pull request by connor-mcshane**
## PR Type  
What kind of change does this PR introduce?  
  
Please check the boxes that apply to this PR.  
  
- [ ] Bugfix  
- [X] Feature  
- [ ] Code style update (formatting, local variables)  
- [ ] Refactoring (no functional changes, no api changes)  
- [ ] Build related changes  
- [ ] CI related changes  
- [ ] Documentation content changes  
- [ ] TranquilityBase application / infrastructure changes  
- [ ] Other... Please describe:  

## Purpose 
1. Added private DNS zone for the shared VPC, so services can be contacted via a domain e.g eagle-console.private.landingzone.com.
2. Added an optional argument to the config script for labelling the bootstrap project with an admin label (something we should be using for development at least).

  
## Reviewers  
 - **@Alex** please review the terraform code.
 - **@Scott** please review the test evidence.  
  
 ## Checklist  
 - [X] This PR is linked to one or more issues.  
 - [X] This PR has been tested (attach evidence if possible).  
 - [X] This PR has passed style guidelines.  
 - [X] This PR has updated the relevant documentation. 

## Testing
- Made a VM (test-ec) in the eagle console project, recorded its IP address in the DNS, enabled ICMP in shared VPC and then pinged  the recorded lookup name.
- 
- 
- 

**NOTE**  
When done, Please use **Squash and Merge** and add a link to the issue number(s) **(#issue-number)** in the merge. This will link the merge to the issue and also make it easier to read through the git log.

### CODE REVIEW

File: tb-gcp-tr/bootstrap/network.tf [hunk 1]
```diff
@@ -64,8 +64,9 @@ resource "google_compute_firewall" "fw_iap_ingress_ssh" {
   }
   description   = "Allows known IAP IP ranges to SSH into VMs"
   name          = "allow-iap-ingress-ssh"
-  network       = google_compute_network.bootstrap_network.name
-  project       = google_project.bootstrap-res.project_id
-  source_ranges = ["35.235.240.0/20"]
+  network = google_compute_network.bootstrap_network.name
+  project = google_project.bootstrap-res.project_id
+  source_ranges = [
+    "35.235.240.0/20"]
```
Comment (alexdgft) [inline tb-gcp-tr/bootstrap/network.tf:70]: @connor-mcshane Can you run a fmt on this?

File: tb-gcp-tr/landingZone/no-itop/input.tfvars [hunk 2]
```diff
@@ -28,23 +28,54 @@ service_projects_number = "3"
...
+    pod_cidr = "10.5.0.0/17",
+    service_cidr = "10.5.128.0/20"
+  }]
+gke_pod_network_name = "gke-pods-snet"
 gke_service_network_name = "gke-services-snet"
 
+#CLOUD DNS
+private_dns_name = "private-landing-zone"
+private_dns_domain_name = "private.landing-zone.com."
+
```
Comment (alexdgft) [inline tb-gcp-tr/landingZone/no-itop/input.tfvars:71]: @connor-mcshane Also fmt on this file

File: tb-gcp-tr/shared-vpc/main.tf [hunk 3]
```diff
@@ -123,3 +121,26 @@ resource "google_compute_shared_vpc_service_project" "service_project" {
...
+###
+# Creating a private DNS
+###
+
+resource "google_dns_managed_zone" "private-zone" {
+  name = var.private_dns_name
+  dns_name = var.private_dns_domain_name
+  project = var.host_project_id
+  description = "Private DNS zone"
+  #labels          = "${var.tags}"
```
Comment (alexdgft) [inline tb-gcp-tr/shared-vpc/main.tf:134]: Remove this line if it's not needed

File: tb-gcp-tr/shared-vpc/main.tf [hunk 4]
```diff
@@ -123,3 +121,26 @@ resource "google_compute_shared_vpc_service_project" "service_project" {
...
+
+###
+# Creating a private DNS
+###
+
+resource "google_dns_managed_zone" "private-zone" {
+  name = var.private_dns_name
+  dns_name = var.private_dns_domain_name
+  project = var.host_project_id
+  description = "Private DNS zone"
```
Comment (alexdgft) [inline tb-gcp-tr/shared-vpc/main.tf:133]: Change to something a bit more descriptive such as "Private DNS zone for Eagle Console"?

File: tb-marketplace/tb-config-creator/tb-config-creator [hunk 5]
```diff
@@ -22,6 +22,7 @@ function print_help {
   echo "  -b, --billing-account-id <string> (REQUIRED) Billing account ID tied to all Tranquility Base consumption"
   echo "  -f, --parent-folder-id <id> Parent folder ID where the Tranquility Base folder will be created"
   echo "  -o, --organization-id <id> Organisation ID where the Tranquility Base folder will be created"
+  echo "  -a, --admin-name <id> (OPTIONAL) admin name to label to bootstrap project"
```
Comment (alexdgft) [inline tb-marketplace/tb-config-creator/tb-config-creator:25]: Is this change supposed to be in this PR?

File: tb-gcp-tr/shared-vpc/examples/advanced/terraform.tfvars [hunk 6]
```diff
@@ -49,4 +49,7 @@ tags = {
     owner = "example owner"
     environment = "dev"
     terraform = "true"
-}
\ No newline at end of file
+}
+
+private_dns_name = "private-landng-zone"
+private_dns_domain_name = "private.landing-zone.com."
```
Comment (alexdgft) [inline tb-gcp-tr/shared-vpc/examples/advanced/terraform.tfvars:55]: In general I'd avoid using domains we don't own. Use local. instead.

File: tb-gcp-tr/bootstrap/network.tf (hunk 1 above)
Comment (connor-mcshane) [reply tb-gcp-tr/bootstrap/network.tf:70]: Sure, do you mean on the whole file or just the code change?

Comment (connor-mcshane) [reply tb-gcp-tr/bootstrap/network.tf:70]: oh whole file I guess :)

File: tb-marketplace/tb-config-creator/tb-config-creator (hunk 5 above)
Comment (connor-mcshane) [reply tb-marketplace/tb-config-creator/tb-config-creator:25]: Oh yeah I just snuck it in, I think I added a point for it in the PR description.

File: tb-gcp-tr/shared-vpc/main.tf (hunk 4 above)
Comment (connor-mcshane) [reply tb-gcp-tr/shared-vpc/main.tf:133]: I think Scott wanted it to be a general private zone. I suppose I could give a description saying that the DNS is for the shared VPC network which would be a better description.

image

File: tb-gcp-tr/shared-vpc/examples/advanced/terraform.tfvars (hunk 6 above)
Comment (connor-mcshane) [reply tb-gcp-tr/shared-vpc/examples/advanced/terraform.tfvars:55]: No worries I changed it to example.com

Comment (alexdgft) [reply tb-gcp-tr/shared-vpc/examples/advanced/terraform.tfvars:55]: Have you committed the change? It's still landing-zone.com

File: tb-marketplace/tb-config-creator/tb-config-creator (hunk 5 above)
Comment (alexdgft) [reply tb-marketplace/tb-config-creator/tb-config-creator:25]: Ok, ideally it would be in a separate PR as it's nothing to do with the Cloud DNS ticket.

Comment (connor-mcshane) [reply tb-marketplace/tb-config-creator/tb-config-creator:25]: Yes ideally.

File: tb-gcp-tr/shared-vpc/examples/advanced/terraform.tfvars (hunk 6 above)
Comment (connor-mcshane) [reply tb-gcp-tr/shared-vpc/examples/advanced/terraform.tfvars]: Just then , sorry got held up on Thursday

File: tb-gcp-tr/landingZone/no-itop/variables.tf [hunk 7]
```diff
@@ -354,3 +354,17 @@ variable "ec_ui_source_bucket" {
   type        = string
 }
 
+#DNS variables
+variable "private_dns_name" {
+  type        = string
+  default     = ""
+  description = "Name for private DNS zone"
```
Comment (alexdgft) [inline tb-gcp-tr/landingZone/no-itop/variables.tf:361]: Can you update these descriptions to match the with itop ones?

File: tb-gcp-tr/landingZone/no-itop/input.tfvars [hunk 8]
```diff
@@ -28,16 +28,47 @@ service_projects_number = "3"
...
+    cidr         = "10.0.5.0/24",
+    pod_cidr     = "10.5.0.0/17",
+    service_cidr = "10.5.128.0/20"
+}]
 gke_pod_network_name     = "gke-pods-snet"
 gke_service_network_name = "gke-services-snet"
 
+#CLOUD DNS
+private_dns_name        = "private-landing-zone"
+private_dns_domain_name = "private.landing-zone.com."
```
Comment (alexdgft) [inline tb-gcp-tr/landingZone/no-itop/input.tfvars:70]: Is this what we're going to be using?

File: tb-gcp-tr/landingZone/no-itop/variables.tf (hunk 7 above)
Comment (connor-mcshane) [reply tb-gcp-tr/landingZone/no-itop/variables.tf:361]: We will remove itop and vault, here is the ticket #286

File: tb-gcp-tr/landingZone/no-itop/input.tfvars (hunk 8 above)
Comment (connor-mcshane) [reply tb-gcp-tr/landingZone/no-itop/input.tfvars:70]: For the moment, can make another ticket to check with the EC team

**Comment by connor-mcshane** [comment]
Closed and made a new PR and branch  _**to remove myself from merge purgatory**_

**Pull request closed without merging**